### PR TITLE
Multiple fixes to FA tests in AMD

### PR DESCRIPTION
**Pull request by remi-or**
This PR fixes a tests across a few models in the following ways:
- solves a multi-device issue in `qwen2_5_omni`
- adds AMD expectations to `gemma3`, `qwen2_5_omni` and `qwen2_5_vl`
- fixes an issue in the `qwen2_5_omni` and `qwen2_5_vl` FA tests: the test changes the `hidden_size` so it breaks compatibility with `mrope`. For this issue, I added a fix using `try` and `except` to ensure the change to `mrope_section` does not propagate to the rest of the tests. I am not sure this is the best way, so tagging @ydshieh
- removed some mutables that were used a default arguments

This PR touches mostly test-related files, so I think it's ok to bundle everything, but please let me know if not. As the non test-related stuff this is mostly multimodal, cc. @zucchini-nlp

### CODE REVIEW

File: src/transformers/models/qwen2_5_omni/modeling_qwen2_5_omni.py [hunk 1]
```diff
@@ -3957,13 +3957,11 @@ def generate(
...
-            torch.tensor([[self.talker.text_eos_token]], dtype=torch.long, device=input_ids.device)
-        )
+        eos_token = torch.tensor([[self.talker.text_eos_token]], dtype=torch.long, device=input_ids.device)
+        eos_embedding = thinker_embed_tokens(eos_token).to(input_ids.device)
 
-        pad_embedding = thinker_embed_tokens(
-            torch.tensor([[self.talker.text_pad_token]], dtype=torch.long, device=input_ids.device)
-        )
+        pad_token = torch.tensor([[self.talker.text_pad_token]], dtype=torch.long, device=input_ids.device)
+        pad_embedding = thinker_embed_tokens(pad_token).to(input_ids.device)
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen2_5_omni/modeling_qwen2_5_omni.py:3966]: oke, we have to move it back to input device, reasonable

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py [hunk 2]
```diff
@@ -594,6 +596,38 @@ def test_get_rope_index_video_with_audio(self):
...
+    def test_flash_attn_2_inference_equivalence(self):
+        # Since the flash_attn_inference_equivalence set hidden_size to 64, we need to double mrope sections
+        previous_rope_scaling = self.model_tester.text_config["rope_scaling"]["mrope_section"][:]
+        self.model_tester.text_config["rope_scaling"]["mrope_section"] = [2, 2, 4]
+        try:
+            self.flash_attn_inference_equivalence(attn_implementation="flash_attention_2", padding_side="left")
+        except Exception as e:
+            self.model_tester.text_config["rope_scaling"]["mrope_section"] = previous_rope_scaling
+            raise e
+        self.model_tester.text_config["rope_scaling"]["mrope_section"] = previous_rope_scaling
```
Comment (zucchini-nlp) [inline tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: This has to be fixed in `update_head_dim` in "test_modeling_common" as in https://github.com/huggingface/transformers/pull/40490/files, since the issue is with the model requiring rope dims to be same as head dim

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py [hunk 3]
```diff
@@ -125,15 +114,33 @@ def __init__(
...
         self.num_channels = num_channels
         self.image_size = image_size
         self.is_training = is_training
         self.vocab_size = vocab_size
         self.num_image_tokens = 32
         self.seq_length = seq_length + self.num_image_tokens
+        # Default vision config is None to avoid a mutable default argument
+        if vision_config is None:
+            vision_config = {
+                "depth": 2,
```
Comment (zucchini-nlp) [inline tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:126]: should not be a problem as long as the `QwenConfig` takes the dict and creates a new `VisionConfig` from it, but agreed that mutable defaults is not a good idea in general

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py (hunk 2 above)
Comment (remi-or) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: Cool to have a fix in the common files! One thing though: here the sum `mrope_section` must be half of `head_dim` I think, so I am not sure we want `the sum of RoPE sections has to be equal to head dim` as is said in the fix. Did the fix work w/ qwen omni?

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py (hunk 3 above)
Comment (remi-or) [reply tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:126]: Yeah it's kind of a nit, I thought it was the problem initially and it was not, but since it's fixed might as well leave it!

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py (hunk 2 above)
Comment (zucchini-nlp) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: right, my bad, poor wording choice 😅

Comment (remi-or) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: np! If the test works on qwen omni I will delete the `mrope` changes, let me know :smile:

Comment (zucchini-nlp) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: Not tested but the arch are similar so it should work, yep

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py (hunk 3 above)
Comment (ydshieh) [reply tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:126]: Let's avoid any mutable type in the arguments 🙏 so keep as it is in this PR

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py [hunk 4]
```diff
@@ -594,6 +596,38 @@ def test_get_rope_index_video_with_audio(self):
 
             self.assertTrue(torch.equal(position_ids, expected_position_ids))
 
+    @require_flash_attn
+    @require_torch_gpu
+    @mark.flash_attn_test
```
Comment (ydshieh) [inline tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: let's `@pytest.mark.flash_attn_test`

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py (hunk 2 above)
Comment (ydshieh) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: If eventually this block has to be kept, you don't need to reset `mrope_section`

```
    def setUp(self):
        self.model_tester = Qwen2_5OmniThinkerForConditionalGenerationTester(self)
        self.config_tester = ConfigTester(self, config_class=Qwen2_5OmniThinkerConfig, has_text_modality=False)
```

the `model_tester` is recreated at the beginning of the test run, it's fresh. 

(unless there is something mutable and persistent being modified)

**Comment by ydshieh** [comment]
Although you are a AMD fans, don't forget to use `run-slow` (unless you feel strong it's not necessary in some cases)

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py (hunk 4 above)
Comment (remi-or) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: This is getting removed so resolving this

File: tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py (hunk 2 above)
Comment (remi-or) [reply tests/models/qwen2_5_omni/test_modeling_qwen2_5_omni.py]: Tested it with the fix, it works, thanks! Removing the inherited test in the two models

**Comment by remi-or** [comment]
run-slow: gemma3, qwen2_5_omni, qwen2_5_vl

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py [hunk 5]
```diff
@@ -672,15 +674,21 @@ def test_small_model_integration_test_batch_wo_image_flashatt2(self):
         # it should not matter whether two images are the same size or not
         output = model.generate(**inputs, max_new_tokens=30)
 
-        EXPECTED_DECODED_TEXT = [
-            'system\nYou are a helpful assistant.\nuser\nWhat kind of dog is this?\nassistant\nThe dog in the picture appears to be a Labrador Retriever. Labradors are known for their friendly and energetic nature, which is evident in',
-            "system\nYou are a helpful assistant.\nuser\nWho are you?\nassistant\n�\n\n addCriterion\nI'm sorry, but I don't understand your question. Could you please provide more context or clarify what you're asking",
```
Comment (remi-or) [inline tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:677]: @zucchini-nlp the second answer seems wrong to me, it used to be `I am Qwen, a large language model created by Alibaba Cloud. I am designed to answer a wide range of questions and provide information on various topics` which makes more sense. I added a #FIXME because it seems out of scope for this PR

**Comment by remi-or** [comment]
run-slow: gemma3, qwen2_5_omni, qwen2_5_vl

File: tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py (hunk 5 above)
Comment (zucchini-nlp) [reply tests/models/qwen2_5_vl/test_modeling_qwen2_5_vl.py:677]: yeah, this is quite similar to the generation from non-FA2 model a few lines above, though I have no idea when it got from `I am Qwen` to `addCriterion`

Thanks, will need to go back in git history to see what changed

**Review by zucchini-nlp — APPROVED** [review]
approving in advance, so it can be merged as long as the slow tests show ✅

**Comment by remi-or** [comment]
run-slow: gemma3, qwen2_5_omni, qwen2_5_vl

**Comment by remi-or** [comment]
run-slow: gemma3, qwen2_5_omni, qwen2_5_vl

**Comment by remi-or** [comment]
There are slow tests failing for
- `gemma3` which are related to #36682 (I think) so not this PR
- `qwen2_5_vl` which has been adressed in comments above and unrelated to this PR
So merging this and adressing them down the line
